modules/andes-core/broker/src/main/java/org/wso2/andes/server/cluster/coordination/rdbms/DatabaseSlotAgent.java

```diff
@@ -25,244 +29,639 @@
+import org.wso2.andes.store.FailureObservingStoreManager;
 
-import java.util.Set;
 import java.util.TreeSet;
+import java.util.Set;
+import java.util.concurrent.ExecutionException;
+
+/**
+ * If the slotManagement is set to "RDBMS" in broker.xml all operations related to slots are handled through this class
+ */
+public class DatabaseSlotAgent implements SlotAgent, StoreHealthListener {
+
+    /**
+     * Indicates and provides a barrier if messages stores become offline.
+     * marked as volatile since this value could be set from a different thread
+     */
+    private volatile SettableFuture<Boolean> messageStoresUnavailable;
+
+    /**
+     * Used for logging purposes
+     */
+    private static Log log = LogFactory.getLog(DatabaseSlotAgent.class);
+
+    /**
+     * Used to perform database operations on the context store.
+     */
+    private AndesContextStore andesContextStore = AndesContext.getInstance().getAndesContextStore();
+
+    /**
+     * Declares the maximum number of times we'll tolerate a store failure
+     */
+    private final int MAX_STORE_FAILURE_TOLERANCE_COUNT = 3;
+
+    public DatabaseSlotAgent() {
+
+        // The existence of messageStoresUnavailable indicates a store failure. Therefore, it should be initialized
+        // with to mark the operational status of the store
+        messageStoresUnavailable = null;
+
+        // Register in the FailureObservingStoreManager to get notifications on store non-operational and operational
+        // status;
+        FailureObservingStoreManager.registerStoreHealthListener(this);
+
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public void createSlot(long startMessageId, long endMessageId, String storageQueueName, String assignedNodeId)
+            throws AndesException {
+
+        String task = "create slot with start message id: " + startMessageId + ", end message id: " + endMessageId
+                      + " for queue: " + storageQueueName + " and node: " + assignedNodeId;
+
+        for (int attemptCount = 1; attemptCount <= MAX_STORE_FAILURE_TOLERANCE_COUNT; attemptCount++) {
+
+            waitUntilStoresBecomeAvailable(task);
+            try {
+                andesContextStore.createSlot(startMessageId, endMessageId, storageQueueName, assignedNodeId);
+                break;
+            } catch (AndesStoreUnavailableException e) {
+                handleFailure(attemptCount, task, e);
+            }
+        }
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public void deleteSlot(String nodeId, String queueName, long startMessageId, long endMessageId)
+            throws AndesException {
+
+        String task = "delete slot with start message id: " + startMessageId + ", end message id: " + endMessageId
+                      + " for queue: " + queueName + " and node: " + nodeId;
+        
+        for (int attemptCount = 1; attemptCount <= MAX_STORE_FAILURE_TOLERANCE_COUNT; attemptCount++) {
+
```

Comment:
Removed all the new lines after the beginning of the method by the PR https://github.com/wso2/andes/pull/403